src/bin/pg_autoctl/cli_config.c

```diff
@@ -110,9 +110,54 @@ cli_config_check(int argc, char **argv)
 
 			(void) cli_config_check_pgsetup(&(mconfig.pgSetup));
 
-			fprintf(stdout, "Current Configuration (includes pgSetup):\n");
-			monitor_config_write(stdout, &mconfig);
-			fprintf(stdout, "\n");
+			if (outputJSON)
+			{
+				JSON_Value *js = json_value_init_object();
```

Comment:
looks like we are going to need more copy/paste s of this part. Would it make sense to enclose creation in a function and keep the variables  in a struct.

Reply:
I'm not sure we can have something generic enough to support the different use cases (direct JSON object use, and nested-object uses, and we will need to add JSON arrays when we add support for `pg_autoctl show events`). I've simplified the output serialisation and memory freeing though. Thanks!